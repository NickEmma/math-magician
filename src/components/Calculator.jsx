import React, { useState } from 'react';
import calculate from '../logic/calculate';
import './Calculator.css';

const Calculator = () => {
  const [state, setState] = useState({
    total: 0,
    next: null,
    operator: null,
  });

  const handleClick = (e) => {
    const result = calculate(state, e.target.textContent);
    setState(result);
  };

  const resultDisplayClick = (e) => {
    setState({
      total: e.target.textContent,
    });
  };

  const { total, next, operation } = state;
  return (
    <>
      <div className="math">
        <h2>Lets do some math!</h2>
      </div>
      <div className="global">
        <div className="answer-span" onChange={resultDisplayClick}>
          {total}
          {operation}
          {next}
        </div>
        <div className="box-one">
          <button
            onClick={handleClick}
            type="button"
            className="btn delete number"
          >
            AC
          </button>
          <button
            onClick={handleClick}
            type="button"
            className="btn  number number"
          >
            +/-
          </button>
          <button
            onClick={handleClick}
            type="button"
            className="btn clear number"
          >
            %
          </button>
          <button
            onClick={handleClick}
            type="button"
            className="btn  equal-to operator"
          >
            รท
          </button>
        </div>
        <div className="box-two">
          <button onClick={handleClick} type="button" className="btn number">
            7
          </button>
          <button onClick={handleClick} type="button" className="btn number">
            8
          </button>
          <button onClick={handleClick} type="button" className="btn number">
            9
          </button>
          <button
            onClick={handleClick}
            type="button"
            className="btn number operator"
          >
            x
          </button>
        </div>
        <div className="box-three">
          <button onClick={handleClick} type="button" className="btn number">
            4
          </button>
          <button onClick={handleClick} type="button" className="btn number">
            5
          </button>
          <button onClick={handleClick} type="button" className="btn number">
            6
          </button>
          <button
            onClick={handleClick}
            type="button"
            className="btn number  operator"
          >
            -
          </button>
        </div>
        <div className="box-four">
          <button onClick={handleClick} type="button" className="btn number">
            1
          </button>
          <button onClick={handleClick} type="button" className="btn number">
            2
          </button>
          <button onClick={handleClick} type="button" className="btn number">
            3
          </button>
          <button
            onClick={handleClick}
            type="button"
            className="btn number operator"
          >
            +
          </button>
        </div>
        <div className="box-five">
          <button
            onClick={handleClick}
            type="button"
            className="btn number zero"
          >
            0
          </button>
          <button
            onClick={handleClick}
            type="button"
            className="btn number dot"
          >
            .
          </button>

          <button
            onClick={handleClick}
            type="button"
            className="btn number operator"
          >
            =
          </button>
        </div>
      </div>
    </>
  );
};
export default Calculator;
