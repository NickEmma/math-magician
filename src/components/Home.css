.home {
  display: flex;
  align-items: center;
  justify-content: center;
  padding-top: 8rem;
}

.home .content {
  flex: 1 1 40rem;
}

.home .content h1 {
  font-size: 2rem;
  color: coral;
  margin-bottom: 2rem;
}

.home .content p {
  font-size: 1.3rem;
  color: #333;
  padding: 1rem 0;
}

.home .image {
  flex: 1 1 40rem;
}

.home .image img {
  width: 100%;
}

@media only screen and (max-width: 700px) {
  .home {
    display: flex;
    flex-direction: column;
  }

  .content {
    margin: 20px 10px;
  }

  .image {
    margin-top: -380px;
  }
}
