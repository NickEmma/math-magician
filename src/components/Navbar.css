.nav {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  background: #333;
  display: flex;
  align-items: center;
  justify-content: space-between;
  padding: 1.2rem 5rem;
}

.nav .logo {
  color: coral;
  font-size: 1.5rem;
  font-style: italic;
}

.nav .items {
  list-style: none;
  display: flex;
  transition: all 0.3s linear;
}

.nav .items li {
  font-size: 1.5rem;
  margin-left: 2rem;
}

.nav .items a {
  color: wheat;
  text-decoration: none;
  font-weight: bold;
}

.nav .items a:hover {
  color: coral;
  transition: 0.3s ease-in;
}
