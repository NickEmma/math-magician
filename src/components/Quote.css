.quoteContainer {
  display: flex;
  flex-direction: column;
  align-items: center;
  height: 80vh;
}

.quotes {
  font-style: italic;
  font-size: 2.2rem;
  width: 50%;
  text-align: center;
  margin: 80px 0 30px 0;
}

.generateQuotes {
  border: none;
  background: orange;
  color: #fff;
  padding: 10px;
  border-radius: 10px;
}

.generateQuotes:hover {
  cursor: pointer;
  background-color: rgb(139, 138, 138);
}
