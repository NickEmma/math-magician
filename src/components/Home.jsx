import React from 'react';
import Image from './images/Mathematics-bro.svg';
import './Home.css';

function Home() {
  return (
    <div className="home">
      <div className="content">
        <h1>Welcome to our page!</h1>
        <p>
          Math Magicians is a website for all fans of mathematics. It is a
        </p>
        <p>
          Single Page App (SPA) that allows users to: Make simple calculations
          and Read a random math-related quotes.
        </p>
      </div>
      <div className="image">
        <img src={Image} alt="math" />
      </div>
    </div>
  );
}

export default Home;
