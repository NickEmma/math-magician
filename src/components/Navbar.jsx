import React from 'react';
import './Navbar.css';
import { Link } from 'react-router-dom';

function Navbar() {
  return (
    <div className="nav">
      <div className="logo">
        <h2>Math Magicians</h2>
      </div>
      <ul className="items">
        <li>
          <Link to="/">Home</Link>
        </li>
        <li>
          <Link to="/calculator">Calculator</Link>
        </li>
        <li>
          <Link to="/quote">Quote</Link>
        </li>
      </ul>
    </div>
  );
}

export default Navbar;
