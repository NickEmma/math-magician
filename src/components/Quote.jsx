import React, { useState } from 'react';
import './Quote.css';

const Quote = () => {
  const [quotes, setQuote] = useState('');

  /* Generate random math quotes */

  const updateQuote = async () => {
    const res = await fetch('https://random-math-quote-api.herokuapp.com/');

    res
      .json()
      .then((res) => {
        setQuote({
          quotes: res.quote,
        });
      })
      .catch();
  };

  /* Convert from object to a string */

  const quote = [JSON.stringify(quotes)].join('').split('');
  quote.pop();
  quote.shift();
  quote.splice(0, 9);

  return (
    <div className='quoteContainer'>
      <div className='quotes'>
        {quotes === '' && (
          <p>&quot;Math is the source code of reality.&quot;</p>
        )}
      </div>
      <p className='quotes'>{quote}</p>
      <button type='button' onClick={updateQuote} className='generateQuotes'>
        Generate quote
      </button>
    </div>
  );
};

export default Quote;
