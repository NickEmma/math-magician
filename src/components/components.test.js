import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Home from './Home';
import Calculator from './Calculator';
import Navbar from './Navbar';
import Quote from './Quote';

it('renders correctly react-test-renderer', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly react-test-renderer', () => {
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly react-test-renderer', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly', () => {
  <Router>
    const tree = renderer.create(
    <Navbar />
    ).toJSON(); expect(tree).toMatchSnapshot();
  </Router>;
});
