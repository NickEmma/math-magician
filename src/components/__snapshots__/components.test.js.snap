// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly react-test-renderer 1`] = `
<div
  className="home"
>
  <div
    className="content"
  >
    <h1>
      Welcome to our page!
    </h1>
    <p>
      Math Magicians is a website for all fans of mathematics. It is a
    </p>
    <p>
      Single Page App (SPA) that allows users to: Make simple calculations and Read a random math-related quotes.
    </p>
  </div>
  <div
    className="image"
  >
    <img
      alt="math"
      src="Mathematics-bro.svg"
    />
  </div>
</div>
`;

exports[`renders correctly react-test-renderer 2`] = `
<div
  className="quoteContainer"
>
  <div
    className="quotes"
  >
    <p>
      "Math is the source code of reality."
    </p>
  </div>
  <p
    className="quotes"
  />
  <button
    className="generateQuotes"
    onClick={[Function]}
    type="button"
  >
    Generate quote
  </button>
</div>
`;

exports[`renders correctly react-test-renderer 3`] = `
Array [
  <div
    className="math"
  >
    <h2>
      Lets do some math!
    </h2>
  </div>,
  <div
    className="global"
  >
    <div
      className="answer-span"
      onChange={[Function]}
    >
      0
    </div>
    <div
      className="box-one"
    >
      <button
        className="btn delete number"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="btn  number number"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="btn clear number"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="btn  equal-to operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="box-two"
    >
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="btn number operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
    </div>
    <div
      className="box-three"
    >
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="btn number  operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="box-four"
    >
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="btn number"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="btn number operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="box-five"
    >
      <button
        className="btn number zero"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="btn number dot"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="btn number operator"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>,
]
`;
